core/thread.c

```diff
@@ -318,3 +318,31 @@ kernel_pid_t thread_create(char *stack, int stacksize, uint8_t priority,
 
     return pid;
 }
+
+static const char *state_names[STATUS_NUMOF] = {
+    [STATUS_STOPPED] = "stopped",
+    [STATUS_ZOMBIE] = "zombie",
+    [STATUS_SLEEPING] = "sleeping",
+    [STATUS_MUTEX_BLOCKED] = "bl mutex",
+    [STATUS_RECEIVE_BLOCKED] = "bl rx",
+    [STATUS_SEND_BLOCKED] = "bl send",
+    [STATUS_REPLY_BLOCKED] = "bl reply",
+    [STATUS_FLAG_BLOCKED_ANY] = "bl anyfl",
+    [STATUS_FLAG_BLOCKED_ALL] = "bl allfl",
+    [STATUS_MBOX_BLOCKED] = "bl mbox",
+    [STATUS_COND_BLOCKED] = "bl cond",
+    [STATUS_RUNNING] = "running",
+    [STATUS_PENDING] = "pending",
+};
+
+#define STATE_NAME_UNKNOWN "unknown"
+
+const char *thread_state_to_string(thread_status_t state)
+{
+    const char *name = state_names[state] ? state_names[state] : NULL;
```

Comment:
shouldn't that be 
```suggestion
    const char *name = state < state_names ? state_names[state] : NULL;
```

Reply:
(this was copied from ps.c unchanged) no, state is an enum, and state_names an array that gets indexed.
But now that I'm taking a closer look this is essentially:

```
if state_names[state] == NULL:
   name = NULL;
else:
  name = state_names[state]
```

So the assignment could just be ```const char *name = state_names[name];```, right?

Or was that check meant for validity of "state"? In that case it must be ```const char *name = state < STATUS_NUMOF ? state_names[state] : NULL;```.

Reply:
Anyhow, I'd like to keep any changes other than the refactoring out of this PR.